tf-modules/archive/provider_credentials_encryption.tf

```diff
@@ -0,0 +1,67 @@
+resource "aws_kms_key" "provider_kms_key" {
+  description = "${var.prefix} Provider credentials encryption"
+  tags        = local.default_tags
+}
+
+data "aws_iam_policy_document" "provider_secrets_encryption" {
+  statement {
+    actions   = [
+      "kms:Encrypt",
+      "kms:Decrypt"
+    ]
+    resources = [aws_kms_key.provider_kms_key.arn]
+  }
+}
+
+resource "aws_iam_role_policy" "provider_secrets_encryption" {
+  role   = aws_iam_role.lambda_api_gateway.id
+  policy = data.aws_iam_policy_document.provider_secrets_encryption.json
+}
+
+resource "aws_lambda_function" "provider_secrets_migration" {
```

Comment:
To state the obvious: these lambdas are only run manually? Are we going to document for users that they need to do this? Or more precisely: when do users need to run this?

Reply:
First question: "yes"

Second question: Yeah, there's still a documentation part of this ticket that needs to be finished. I wanted to make sure that it all made sense and that there weren't going to be any major changes before I wrote the user docs / changelog.

Reply:
OK. I'm fine with docs being a separate ticket but I'd like to see the CHANGELOG entries in this PR. I think it's nice for historical review and it's also easier imo to include the CHANGELOG entries in the PR with the actual changes because it reminds me what changed

Reply:
I'm going to include the Changelog entries and migration instructions in this PR, I just wanted to make sure we were all in agreement with the changes before I bothered writing the migration instructions.

Reply:
Added migration notes in the Changelog